### PR TITLE
docs: More progress towards typing

### PR DESCRIPTION
**Pull request by silasary**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition

## Description
You know the drill.  It's more progress towards 100% type annotation coverage

## Changes
- Also fixed a couple instances of B007 (unused loop variable) and C816 (missing trailing comma)

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

File: dis_snek/models/discord_objects/user.py [hunk 1]
```diff
@@ -465,17 +465,17 @@ async def timeout(
...
+    async def kick(self, reason: Absent[str] = MISSING) -> None:
         """
         Remove a member from the guild.
 
         Args:
             reason: The reason for this removal
 
         """
-        return await self._client.http.remove_guild_member(self._guild_id, self.id)
+        await self._client.http.remove_guild_member(self._guild_id, self.id, reason=reason)
```
Comment (benwoo1110) [inline dis_snek/models/discord_objects/user.py:476]: ~~not really a typing fix here~~

Comment (silasary) [reply dis_snek/models/discord_objects/user.py:476]: Technically it is.

Or rather, mypy raises it as a warning when you explicitly return the value of a function that always returns None.

Comment (benwoo1110) [reply dis_snek/models/discord_objects/user.py:476]: Oh I was referring to the reason=reason

Comment (silasary) [reply dis_snek/models/discord_objects/user.py:476]: Oh, right, I forgot that I fixed that :D

**Comment by LordOfPolls** [comment]
This PR has been merged. @silasary if youre done with that branch please delete it

**Comment by silasary** [comment]
I'm done with it.... For Now...

**Pull request merged**